### PR TITLE
create joystick for mob version

### PR DESCRIPTION
**Pull request by 19thanks96**
#181 
- [x] delete the old button
- [x] create a joystick
- [x] add code to change the speed of the fighter when changing the nipple location

### CODE REVIEW

File: src/step.js [hunk 1]
```diff
@@ -22,6 +23,34 @@ export function Step() {
...
 
+  if(mobileVelocity.createdJoystick) {
+    mobileTrafficControl()
+  }
+
   const lastEnemy = enemies[enemies.length - 1]
   levelOverIfLastEnemyOut(lastEnemy, gameState)
 }
+
+function mobileTrafficControl() {
```
Comment (DarthVanger) [inline src/step.js:34]: I think we could have this function in another file.
We can create a folder `src/mobile/` and create a file for this function.
Also we could rename it to something like `setAirfighterVelocityFromMobileJoystick`.

File: src/mobileJoystick.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    y: 0, 
+    left: false,
+    right: false,
+    up: false,
+    down: false,
+}
+
+export function createJoystick() {
+
+svg.setAttribute('width', svgWidth);
```
Comment (DarthVanger) [inline src/mobileJoystick.js:23]: The code inside the function should be indented by 2 spaces.

File: src/step.js [hunk 3]
```diff
@@ -22,6 +23,34 @@ export function Step() {
   renderRocket()
   airfighter.render()
 
+  if(mobileVelocity.createdJoystick) {
```
Comment (DarthVanger) [inline src/step.js:26]: It sounds weird that velocity has `.createdJoystick`.
Maybe `mobileJoystick` would be a better name for this object?
And also `createdJoystick` could become `if (mobileJoystick.isShown)`.

Secondly, we could also have something like `if (isMobileDevice)` instead.
Maybe this way the logic would be easier to understand.

File: src/mobileJoystick.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    } else {
+        const scaleFactor = maxDistance / distanceToCenter;
+        distanceFromCenterX *= scaleFactor
+        distanceFromCenterY *= scaleFactor
+        distanceToCenter = Math.sqrt(distanceFromCenterX * distanceFromCenterX + distanceFromCenterY * distanceFromCenterY)
+        
+        circleInside.setAttribute('cx', distanceFromCenterX + joystickRadius);
+        circleInside.setAttribute('cy', distanceFromCenterY + joystickRadius);
+        
+        mobileVelocity.x = distanceFromCenterX / maxDistance
```
Comment (DarthVanger) [inline src/mobileJoystick.js:84]: Looks like `mobileVelocity.x` is position of joystick, not velocity.
Maybe `mobileJoystick.x` would sound better

File: src/step.js (hunk 1 above)
Comment (19thanks96) [reply src/step.js:34]: [x]

File: src/mobileJoystick.js (hunk 2 above)
Comment (19thanks96) [reply src/mobileJoystick.js:23]: - [x]

File: src/step.js (hunk 3 above)
Comment (19thanks96) [reply src/step.js:26]: - [x] finished

File: src/mobileJoystick.js (hunk 4 above)
Comment (19thanks96) [reply src/mobileJoystick.js:84]: - [x] finished

File: src/mobile/changeMobileSpeed.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    airfighter.isKeyUpPressed = true
+    airfighter.vy *= isMobileDevice.y * -1
+  } else if (isMobileDevice.y > 0) {
+    airfighter.isKeyDownPressed = true
+    airfighter.vy *= isMobileDevice.y
+  } else {
+    airfighter.isKeyUpPressed = false
+    airfighter.isKeyDownPressed = false
+  }
+}
```
Comment (DarthVanger) [inline src/mobile/changeMobileSpeed.js:26]: I would rename `changeMobileSpeed.js` -> `mobileVelocityController.js`

File: src/mobileJoystick.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
+// Create an SVG element
+const svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg')
+const circleInside = document.createElementNS(
```
Comment (DarthVanger) [inline src/mobileJoystick.js:3]: `circleInside` -> `innerCircle` :)

File: src/mobileJoystick.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+  'circle'
+)
+
+// Set the width as a percentage of the screen size
+const screenWidth = window.innerWidth
+const svgWidth = (20 / 100) * screenWidth
+const joystickRadius = svgWidth / 2
+const joystickNippleRadius = svgWidth / 8
+
+export const isMobileDevice = {
```
Comment (DarthVanger) [inline src/mobileJoystick.js:14]: Please rename `isMobileDevice` -> `mobileJoystick`.

`isMobileDevice` is a name for a boolean value, which would tell if user is playing from mobile device or not.
It was for another idea, sorry for confusion :)

**Review by DarthVanger — APPROVED** [review]

**Review by Gehinom — COMMENTED** [review]
LGTM

**Pull request merged**